src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -0,0 +1,278 @@
+        basic_info_attributes = basic_info_cluster.Attributes
+
+        # Preconditions
+        self.step("0")
+
+        logging.info("Ensuring DUT is commissioned to TH")
+
+        # Confirms commissioning of DUT on TH as it reads basic info
+        await self._read_attribute_expect_success(
+          kRootEndpointId,
+          basic_info_cluster,
+          basic_info_attributes.FeatureMap,
+          self.dut_node_id
+        )
+
+        logging.info("Ensuring ICD is commissioned to TH")
+
+        # Confirms commissioning of ICD on TH as it reads basic info
+        await self._read_attribute_expect_success(
+          kRootEndpointId,
+          basic_info_cluster,
+          basic_info_attributes.FeatureMap,
+          self.icd_nodeid
+        )
+
+        self.step("1a")
+
+        idle_mode_duration = await self._read_attribute_expect_success(
+          kICDMEndpointId,
+          icdm_cluster,
+          icdm_attributes.IdleModeDuration,
+          self.icd_nodeid
+        )
+        logging.info(f"IdleModeDuration: {idle_mode_duration}")
+
+        active_mode_duration = await self._read_attribute_expect_success(
+          kICDMEndpointId,
+          icdm_cluster,
+          icdm_attributes.ActiveModeDuration,
+          self.icd_nodeid
+        )
+        logging.info(f"ActiveModeDuration: {active_mode_duration}")
+
+        self.step("1b")
+
+        # Subscription to ActiveChanged
+        event = brb_info_cluster.Events.ActiveChanged
+        self.q = queue.Queue()
+        urgent = 1
+        cb = SimpleEventCallback("ActiveChanged", event.cluster_id, event.event_id, self.q)
+        subscription = await self.default_controller.ReadEvent(nodeid=self.dut_node_id, events=[(kICDBridgedEndpointId, event, urgent)], reportInterval=[1, 3])
+        subscription.SetEventUpdateCallback(callback=cb)
+
+        stay_active_duration = 1000
+        logging.info(f"Sending KeepActiveCommand({stay_active_duration}ms)")
+        self._send_keep_active_command(stay_active_duration)
+
+        logging.info("Waiting for ActiveChanged from DUT...")
+        promised_active_duration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, stay_active_duration))/1000)
+
+        asserts.assert_true(promised_active_duration >= stay_active_duration, "PromisedActiveDuration < StayActiveDuration")
+
+        self.step("2")
+
+        stay_active_duration = 1500
+        logging.info(f"Sending KeepActiveCommand({stay_active_duration}ms)")
+        self._send_keep_active_command(stay_active_duration)
+
+        logging.info("Waiting for ActiveChanged from DUT...")
+        promised_active_duration = await self._wait_for_active_changed_event((idle_mode_duration + max(active_mode_duration, stay_active_duration))/1000)
+
+        # wait for active time duration
+        time.sleep(max(stay_active_duration/1000, active_mode_duration))
+        # ICD now should be in idle mode
+
+        # sends 3x keep active commands
```

Comment:
Is this is a LIT device? If so, don't you need to wait for the check in to send a new command?

Reply:
In this case I want to send KeepActive commands to DUT precisely when they ICD is idle. My understanding is that DUT then should send a single ActiveChanged event back to TH whenever ICD becomes active. I think that was the spirit of the test proposed

Reply:
is it? My understanding was that it shouldn't send back multiple active changed events if it actually receives the commands. I don't think an idle device will actually process these. Can you please file a follow up issue to check this behaviour with the ICD team to ensure it is a) testing what needs to be tested and b) works with how the ICD devices work?

Reply:
Follow up is OK here for TE. Please link the issue into this test plan before closing this issue. Then you can close. I'll approve this now on that condition.